test/surface_test.py

```diff
@@ -3133,6 +3133,46 @@ def test_overlap_check(self):
                 surf.blit(surf, (d_x, d_y), (s_x, s_y, 50, 50))
                 self.assertEqual(surf.get_at(test_posn), rectc_right)
 
+    def test_overlap_check_unrolled(self):
```

Comment:
We can remove this unrolled test now?

It'd be great if we had paramatised tests. So for situations like this we can see where it fails. Unfortunately unit test doesn't have this feature until newer versions. Perhaps there is a good polyfill for old versions around however.
See subTest in python 3.4 https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests I started an issue for paramatized tests here: https://github.com/pygame/pygame/issues/2088